robin/model/multimodal_encoder/builder.py

```diff
@@ -1,30 +1,27 @@
-import os
 from .clip_encoder import CLIPVisionTower
 from .open_clip import OpenCLIPVisionTower
 from .timm_vision import TimmVisionTower
-from .encoder_info import CLIP_COMPATIBLE_MODELS, OPENCLIP_CONFIG_MAP
+
+
+OPEN_CLIP_MODELS = [
+    'ViT-B-16/laion2b_s34b_b88K',
+    'ViT-L-14/laion2B-s32B-b82K',
+    'ViT-H-14/laion2B-s32B-b79K',
+    'ViT-g-14/laion2B-s34B-b88K',
+]
+
 
 def build_vision_tower(vision_tower_cfg, **kwargs):
     vision_tower = getattr(vision_tower_cfg, 'mm_vision_tower', getattr(vision_tower_cfg, 'vision_tower', None))
-    
-    if os.path.exists(vision_tower):
-        name = vision_tower.split("/")[-1].lower()
-
-        if name in CLIP_COMPATIBLE_MODELS:
-            return CLIPVisionTower(vision_tower, args=vision_tower_cfg, **kwargs)
-        elif "dino" in str(vision_tower).lower():
-            return TimmVisionTower(vision_tower, args=vision_tower_cfg, **kwargs)
-        elif name in OPENCLIP_CONFIG_MAP.keys():
-            return OpenCLIPVisionTower(vision_tower, args=vision_tower_cfg, **kwargs)
-        else:
-            print(f'Local model not handled in configs (might crash): {vision_tower}')
-            return OpenCLIPVisionTower(vision_tower, args=vision_tower_cfg, **kwargs)
+
+    if vision_tower in OPEN_CLIP_MODELS:
+        return OpenCLIPVisionTower(vision_tower, args=vision_tower_cfg, **kwargs)
+
+    elif vision_tower.lower().startswith("openai"):
+        return CLIPVisionTower(vision_tower, args=vision_tower_cfg, **kwargs)
+
+    elif vision_tower.lower().startswith("timm"):
```

Comment:
We need a better way of doing this - let's think of one and add it. The 'timm' vision tower is special because it's for pure timm models, like Dino, as opposed to Open-Clip timm models, which are still through Timm, but some strange Open-Clip timm hybrid setup.

Reply:
Add new arg called `vision type`, parse this arg to select the visiontower class.
and create a outside sheet to map each vision tower name to its type.

Reply:
Pls check this commit for add `vision_tower_type`
https://github.com/AGI-Collective/Robin/commit/e04d449459389f566433ff88344380ad06a5a8ca